### ISSUE TITLE
macOS Sierra

### ISSUE THREAD

**Issue by gklka** [davekeck/DisableExtremeScrollAcceleration#3]
It is that time of the year: Worked fine on my El Capitan, but refuses to work on macOS Sierra GM. Any hopes that you can fix it?

**Comment by esotericsean** [comment]
I'm hoping for a fix, too!

**Comment by Joostdehaas** [comment]
Me too! Hope someone can work on it. (I made an account just to say that...)

**Comment by tekkub** [comment]
Chiming in with another "me too!"

**Comment by jonasmellquist** [comment]
I want to be part of this choir too :)

**Comment by gklka** [comment]
I've looked into the source code, and it compiles very well on Sierra. It seems also running, but does not do anything. The property which it adjusts (`HIDScrollCountMinDeltaToStart`) is a valid key in the `/System/Library/Extensions/IOHIDFamily.kext/Contents/Info.plist`. I don't know what causes the problem.

**Comment by cenkdominic** [comment]
+1

**Comment by qwazix** [comment]
Seems that it has something to do with *mac system integrity protection*. I tried editing the plist by hand but it is not possible.

**Comment by gklka** [comment]
You can disable SIP from recovery boot (⌘R during boot), but I think that does not help.

**Comment by allanjackson** [comment]
Yeah, even disabling SIP, editing the values seems to have no effect.

**Comment by jryom** [comment]
I think the problem is that Apple has deprecated the API that DisableExtremeScrollAcceleration depended upon, namely IOHID. 

You can read more on the issue [here](http://dae.me/blog/2144/on-macos-sierra-support-and-the-future-of-smoothmouse/). Doesn't look like there is much hope for this method under MacOS Sierra unfortunately.

**Comment by esotericsean** [comment]
So, just in case someone still needs a workaround. I found that installing USB Overdrive and disabling Acceleration (lower left) fixes the issue.

<img width="780" alt="screen shot 2016-11-21 at 11 25 48 am" src="https://cloud.githubusercontent.com/assets/17632881/20497271/6364110c-afdd-11e6-97be-e5554397855f.png">

**Comment by gklka** [comment]
So does this command:
`defaults write .GlobalPreferences com.apple.scrollwheel.scaling -1` (you have to relogin after that)

But both of the methods work only for mouses, not for the trackpad :(

**Comment by esotericsean** [comment]
Hmm, just tested it by scrolling through my hotbar in Minecraft and USB Overdrive definitely fixes the trackpad scrolling there. What app(s) are you having trouble using with scrolling with a trackpad?

**Comment by jryom** [comment]
According to [USB Overdrive's website](http://www.usboverdrive.com/USBOverdrive/Devices.html), the built-in trackpad is not affected by their software:

> **Apple Magic Mouse, Magic Trackpad and Magic Keyboard**
Apple's built-in drivers use undocumented private APIs to forward multitouch gestures to the OS. I've been talking to Apple for a long time about this, but it looks like multitouch event generation is not going to be available to 3rd parties anytime soon. Since multitouch gestures are more important than any other additional setting, the USB Overdrive is now leaving all Apple devices alone and no longer tries to handle them in any way.

**Comment by sigsergv** [comment]
This is not a solution or workaround but I managed to reduce acceleration to more or less acceptable values by settings scrolling speed to 0.08:

`defaults write .GlobalPreferences com.apple.trackpad.scrolling 0.08`

Scrolling is slow but acceleration is much lesser too. Don't forget to logout after changing that value.
